Detections/AzureDevOps/AzDOServiceConnectionUsage.yaml

```diff
@@ -31,11 +31,15 @@ query: |
   //"122", "Build", "ProjectB"
   ];
   AzureDevOpsAuditing
-  | where TimeGenerated > ago(timeframe)
+  | where TimeGenerated >= ago(timeframe)
   | where OperationName == "Library.ServiceConnectionExecuted" 
   | extend DefId = tostring(Data.DefinitionId), Type = tostring(Data.PlanType), ConnectionId = tostring(Data.ConnectionId)
   | parse ScopeDisplayName with OrganizationName ' (Organization)'
-  | summarize CurrentCount = dcount(tostring(ConnectionId)), ConnectionNames = make_set(tostring(Data.ConnectionName)), StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by OrganizationName, tostring(DefId), tostring(Type), ProjectId, ProjectName
+  | summarize CurrentCount = dcount(tostring(ConnectionId)), ConnectionNames = make_set(tostring(Data.ConnectionName)), StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) 
+    by OrganizationName, tostring(DefId), tostring(Type), ProjectId, ProjectName
   | where CurrentCount > ServiceConnectionThreshold
   | join kind=anti BypassDefIds on $left.DefId==$right.DefId and $left.Type == $right.Type and $left.ProjectName == $right.ProjectName
-  | extend link = iif(Type == "Build", strcat('https://dev.azure.com/', OrganizationName, '/', ProjectName, '/_build?definitionId=', DefId) ,strcat('https://dev.azure.com/', OrganizationName, '/', ProjectName, '/_release?_a=releases&view=mine&definitionId=', DefId)), timestamp = StartTime
\ No newline at end of file
+  | extend link = iif(
+    Type == "Build", strcat('https://dev.azure.com/', OrganizationName, '/', ProjectName, '/_build?definitionId=', DefId),
+    strcat('https://dev.azure.com/', OrganizationName, '/', ProjectName, '/_release?_a=releases&view=mine&definitionId=', DefId))
+  | extend timestamp = StartTime
```

Comment:
Any other entities we can bring apart from just timestamp 
e.g. AccountCustomEntity = ActorUPN, IPCustomEntity = IpAddress  ?

Reply:
Not on this one, I thought of this too, but nope.